trackpy/feature.py

```diff
@@ -532,71 +506,97 @@ def batch(frames, diameter, minmass=100, maxsize=None, separation=None,
+            source=source,
+            # Use default values of `locate`...
+            minmass=None, maxsize=None, separation=None, noise_size=1,
+            smoothing_size=None, threshold=None, invert=False, percentile=64,
+            topn=None, preprocess=True, max_iterations=10, filter_before=None,
+            filter_after=None, characterize=True, engine='auto'
+        )
+        # ... and overwrite with provided keyword arguments
+        meta_info.update(kwargs)
+
         if isinstance(meta, six.string_types):
             with open(meta, 'w') as file_obj:
                 record_meta(meta_info, file_obj)
         else:
             # Interpret meta to be a file handle.
             record_meta(meta_info, meta)
 
-    all_features = []
-    for i, image in enumerate(frames):
-        features = locate(image, diameter, minmass, maxsize, separation,
-                          noise_size, smoothing_size, threshold, invert,
-                          percentile, topn, preprocess, max_iterations,
-                          filter_before, filter_after, characterize,
-                          engine)
-        if hasattr(image, 'frame_no') and image.frame_no is not None:
-            frame_no = image.frame_no
-            # If this works, locate created a 'frame' column.
-        else:
-            frame_no = i
-            features['frame'] = i  # just counting iterations
-        logger.info("Frame %d: %d features", frame_no, len(features))
-        if len(features) == 0:
-            continue
+    # Prepare wrapped function for mapping to `frames`
+    curried_locate = partial(locate, **kwargs)
 
-        if output is None:
-            all_features.append(features)
-        else:
-            output.put(features)
+    # Handle & validate argument `processes`
+    if processes == "auto":
+        processes = None  # Is replaced with `os.cpu_count` in Pool
+    elif not isinstance(processes, six.integer_types):
+        raise TypeError("`processes` must either be an integer or 'auto', "
+                        "was type {}".format(type(processes)))
+
+    if processes is None or processes > 1:
+        # Use multiprocessing
+        pool = Pool(processes=processes)
+        map_func = pool.imap
+    else:
+        pool = None
+        map_func = map
+
+    if after_locate is None:
+        def after_locate(frame_no, features):
+            return features
+
+    try:
+        all_features = []
+        for i, features in enumerate(map_func(curried_locate, frames)):
+            image = frames[i]
+            if hasattr(image, 'frame_no') and image.frame_no is not None:
+                frame_no = image.frame_no
+                # If this works, locate created a 'frame' column.
+            else:
+                frame_no = i
+                features['frame'] = i  # just counting iterations
+            features = after_locate(frame_no, features)
+
+            logger.info("Frame %d: %d features", frame_no, len(features))
```

Comment:
I intended this to be in the default `after_locate`. What do you think @lagru and @nkeim?

Reply:
That was my original intention as well. But I think this approach is actually more flexible and has the least surprises. From the user perspective: its not intuitive that if I supply my own post-processing function suddenly unrelated behavior (logging) is changed. At least that's my gut feeling. This way, logging and post-processing can be configured independently.

Otherwise the docstring should explicitly mention that providing `after_locate` overwrites the logging functionality.

Reply:
That makes sense to me. Users could always control the logging through the logging module.